pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -431,6 +435,74 @@ func (moa *MultiOrgAlertmanager) AlertmanagerFor(orgID int64) (Alertmanager, err
 	return orgAM, nil
 }
 
+// CreateSilence creates a silence in the Alertmanager for the organization provided, returning the silence ID. It will
+// also persist the silence state to the kvstore immediately after creating the silence.
+func (moa *MultiOrgAlertmanager) CreateSilence(ctx context.Context, orgID int64, ps *alertingNotify.PostableSilence) (string, error) {
+	moa.alertmanagersMtx.RLock()
+	defer moa.alertmanagersMtx.RUnlock()
+
+	orgAM, existing := moa.alertmanagers[orgID]
+	if !existing {
+		return "", ErrNoAlertmanagerForOrg
+	}
+
+	if !orgAM.Ready() {
+		return "", ErrAlertmanagerNotReady
+	}
+
+	// Need to create the silence in the AM first to get the silence ID.
+	silenceID, err := orgAM.CreateSilence(ctx, ps)
+	if err != nil {
+		return "", err
+	}
+
+	return silenceID, moa.updateSilenceState(ctx, orgAM, orgID)
```

Comment:
Do we want to return an error if the silence has been created but couldn't be persisted? Persistence is technically not necessary for the silence to be applied, just to ensure it's not lost across restarts.

Reply:
Fixed: f4aa99a228a74c47c39871dc545124ba3ffcbf44